src/clj/puppetlabs/services/jruby/jruby_puppet_core.clj

```diff
@@ -105,6 +108,25 @@
     (update-in [:max-active-instances] #(or % (default-pool-size (ks/num-cpus))))
     (update-in [:max-requests-per-instance] #(or % 0))))
 
+(def facter-jar
+  "Well-known name of the facter jar file"
+  "facter.jar")
+
+(schema/defn ^:always-validate
+  add-facter-jar-to-system-classloader
+  "Searches the ruby load path for a file whose name matches that of the
+  facter jar file.  The first one found is added to the system classloader's
+  classpath.  If no match is found, an info message is written to the log
+  but no failure is returned"
+  [ruby-load-path :- (schema/both (schema/pred vector?) [schema/Str]) ]
+  (if-let [facter-jar (first
+                        (filter fs/exists?
+                          (map #(fs/file % facter-jar) ruby-load-path)))]
+    (do
+      (log/debugf "Adding facter jar to classpath from: %s" facter-jar)
+      (ks-classpath/add-classpath facter-jar))
```

Comment:
I've seen this same technique used on the Java-end for things like dynamically loading up a JDBC jar from the filesystem. I honestly have more beef with the Java API for making us do this dumb stuff than actually doing the dumb stuff ourselves. So, this looks good to me.